spec.html

```diff
@@ -910,6 +910,22 @@ <h1>Invoking Syntax-Directed Operations</h1>
         <p>Regardless of the convention, the invocation causes the execution of the algorithm that the syntax-directed operation associates (either explicitly or implicitly, see <emu-xref href="#sec-implicit-definitions-for-chain-productions"></emu-xref>) with the subject Parse Node's production. Thus, within any algorithm of the syntax-directed operation, the subject Parse Node is necessarily an instance of one of the production(s) associated with that algorithm.</p>
       </emu-clause>
 
+      <emu-clause id="sec-referring-to-parse-nodes">
+        <h1>Referring to Parse Nodes</h1>
+        <p>Although a syntax-directed operation is always invoked with a Parse Node, the heading for a syntax-directed operation does not declare a parameter to receive it. Instead, each definition uses various phrases to refer to the subject Parse Node and its children, depending on the production(s) associated with that definition. The general idea is that the algorithm refers to nonterminals in the associated production(s), and thereby refers to the Parse Nodes that correspond to those nonterminals for a particular invocation.</p>
+        <p>To refer to the subject Parse Node:</p>
+        <ul>
+          <li>If all of the associated productions have the same left-hand side nonterminal, then that nonterminal is used with “this” (e.g., “this |ExportDeclaration|”).</li>
+          <li>Otherwise, “this Parse Node” is used.</li>
+        </ul>
+        <p>To refer to the children of the subject Parse Node: In practice, algorithms only reference children that are instances of nonterminals.</p>
+        <ul>
+          <li>If each associated production has exactly one right-hand occurrence of the nonterminal, then the algorithm can refer to it with “the” or without a prefix (e.g., “the |StatementList|” or “|StatementList|”). If the same nonterminal also appears on the LHS of the production, the algorithm can also use “the derived” (e.g., “the derived |UnaryExpression|”) to emphasize a reference to the right-hand nonterminal.</li>
+          <li>If a production has multiple right-hand occurrences of the nonterminal, they are distinguished by ordinals (e.g., “the first |DecimalDigits|” or “the fourth |HexDigit|”).</li>
+          <li>If a production has an optional nonterminal (e.g., “|StatementList?|”), the algorithm must check if the corresponding Parse Node is present before using it (e.g., “If the |StatementList| is present, return the VarDeclaredNames of |StatementList|.”)</li>
```

Comment:
```suggestion
          <li>If a production has an optional nonterminal (e.g., “|StatementList?|”), the algorithm must check if the corresponding Parse Node is present before using it (e.g., “If |StatementList| is present, return VarDeclaredNames of |StatementList|.”)</li>
```

The first "the" should be avoided. The second "the" we're really inconsistent about, but I'd prefer to not encourage it with the example.

Reply:
> The first "the" should be avoided.

I'm pretty sure that hasn't been established. (If it has, please link.)

I'm inclined to the opposite opinion: eliding "the" before a nonterminal should be avoided.

Alternatively, we could use a scheme of superscripts such as you suggested in issue #1768. The editors might want to decide on that before this PR goes ahead, to avoid churn.

---

> The second "the" we're really inconsistent about, but I'd prefer to not encourage it with the example.

Okay, done.

Would you like a PR that removes "the" from before the SDO-name in invocations? There's about 400 occurrences.

Reply:
> I'm pretty sure that hasn't been established. (If it has, please link.)

It hasn't.

> Alternatively, we could use a scheme of superscripts such as you suggested in issue #1768. The editors might want to decide on that before this PR goes ahead, to avoid churn.

Nah, even if we eventually went with that eventually, I'd prefer to first do this intermediate step.

> Would you like a PR that removes "the" from before the SDO-name in invocations? There's about 400 occurrences.

If you don't mind, yes. That would help us a lot in discussing the change.